src/sagemaker/config/config_schema.py

```diff
@@ -963,6 +1028,31 @@ def _simple_path(*args: str):
                     TYPE: OBJECT,
                     ADDITIONAL_PROPERTIES: False,
                     PROPERTIES: {
+                        DEBUG_HOOK_CONFIG: {
+                            "anyOf": [
+                                {TYPE: "boolean"},
```

Comment:
Supporting this would make this config deviate from the public API. Would be better to keep this aligned to the API